frontends/web/src/routes/account/summary/accountssummary.tsx

```diff
@@ -65,9 +68,73 @@ export const AccountsSummary = ({
   const [accountsTotalBalance, setAccountsTotalBalance] = useState<accountApi.TAccountsTotalBalance>();
   const [coinsTotalBalance, setCoinsTotalBalance] = useState<accountApi.TCoinsTotalBalance>();
   const [balances, setBalances] = useState<Balances>();
+  const { lightningConfig } = useLightning();
 
   const hasCard = useSDCard(devices);
 
+
+  const [lightningBalance, setLightningBalance] = useState<accountApi.IBalance>();
+  const fetchLightningBalance = useCallback(async () => {
+    setLightningBalance(await getLightningBalance());
+  }, []);
+
+  const getCoinsTotalBalance = useCallback(async () => {
+    try {
+      const coinBalance = await accountApi.getCoinsTotalBalance();
+      if (!mounted.current) {
+        return;
+      }
+      setCoinsTotalBalance(coinBalance);
+    } catch (err) {
+      console.error(err);
+    }
+  }, [mounted]);
+
+  // if there is an active lightning account derived from this keystore, subscribe to any node state changes.
+  useEffect(() => {
+    const lightningAccounts = lightningConfig.accounts;
+    if (lightningAccounts.length) {
+      fetchLightningBalance();
+      const subscriptions = [
+        subscribeNodeState(async () => {
+          await fetchLightningBalance();
+          await getCoinsTotalBalance();
+        })
+      ];
+      return () => unsubscribe(subscriptions);
+    }
+  }, [fetchLightningBalance, getCoinsTotalBalance, lightningConfig]);
+
+  const lightningKeystoreName = useLoad(() => {
+    const account = lightningConfig.accounts[0];
+    return account
+      ? accountApi.keystoreName(account.rootFingerprint).then(response => response.keystoreName ?? '')
+      : Promise.resolve('');
+  }, [lightningConfig]) || '';
+
+  // lightning account exists but is not from any connected or remembered keystores
+  const hasLightningFromOtherKeystore = (
+    lightningConfig.accounts.length !== 0
+    && (
+      !accountsByKeystore.some(({ keystore }) => {
+        return keystore.rootFingerprint === lightningConfig.accounts[0].rootFingerprint;
+      })
+    )
+  );
+  let keystores = accountsByKeystore.length;
+  let keystoreDisambiguatorName;
+  if (hasLightningFromOtherKeystore) {
+    keystores += 1;
+    keystoreDisambiguatorName = isAmbiguousName(
+      lightningKeystoreName,
+      lightningKeystoreName
+        ? [...accountsByKeystore, { keystore: { name: lightningKeystoreName } }]
+        : accountsByKeystore
```

Comment:
You are computing this several times: here and for each `SummaryBalance`. You could simplify and make it more efficient by reusing it, right?

Reply:
I see, ok I will update it

Reply:
I created a new variable `allKeystores` which allowed me to get rid of `keystores` var and simplify the usage of `isAmbiguousName`